### PR TITLE
[feature][connector-file-local] add save mode function for localfile

### PR DESCRIPTION
**Pull request by chl-wxp**
### Purpose of this pull request
add save mode function for localfile

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?
add unit test

### Check list

* [x] add save mode function for localfile

### CODE REVIEW

**Comment by chl-wxp** [comment]
add doc

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-local-e2e/src/test/resources/json/fake_to_local_file_json.conf [hunk 1]
```diff
@@ -77,5 +77,7 @@ sink {
     file_format_type = "json"
     filename_time_format = "yyyy.MM.dd"
     is_enable_transaction = true
+    "schema_save_mode"="RECREATE_SCHEMA"
+    "data_save_mode"="DROP_DATA"
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-local-e2e/src/test/resources/json/fake_to_local_file_json.conf:81]: Please check whether savemode is effective

**Comment by Hisoka-X** [comment]
cc @TyrantLucifer as well.

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-local-e2e/src/test/resources/json/fake_to_local_file_json.conf (hunk 1 above)
Comment (chl-wxp) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-local-e2e/src/test/resources/json/fake_to_local_file_json.conf:81]: added

**Review by Hisoka-X — APPROVED** [review]
LGTM.

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**